Sources/SKCore/ToolchainRegistry.swift

```diff
@@ -157,10 +173,30 @@ extension ToolchainRegistry {
   }
 
   func _registerToolchain(_ toolchain: Toolchain) throws {
-    guard toolchainIdentifiers[toolchain.identifier] == nil else {
-      throw Error.duplicateToolchainIdentifier
+    // Only allow non-equivalent duplicates for the XcodeDefault toolchain identifier.
+    if toolchain.identifier == ToolchainRegistry.xcodeDefaultToolchainIdentifier {
+      if let xcodeToolchains = toolchainsByIdentifier[toolchain.identifier] {
+        guard !xcodeToolchains.contains(toolchain) else {
+          throw Error.duplicateToolchainIdentifier
```

Comment:
I suggest splitting the error up to cover the path case separately from the identifier case.

Reply:
Done